packages/sdks/typescript/src/index.ts

```diff
@@ -1,163 +1,212 @@
+  response: ChainCallResponseDto
 }
 type StreamResponseCallbacks = {
-  onMessage?: (chainEvent: ChainEvent) => void
+  onEvent?: ({
+    event,
+    data,
+  }: {
+    event: StreamEventTypes
+    data: ChainEventDto
+  }) => void
   onFinished?: (data: StreamChainResponse) => void
   onError?: (error: Error) => void
 }
 
 export class LatitudeSdk {
-  private latitudeApiKey: string
+  private projectId?: number
+  private apiKey: string
   private routeResolver: RouteResolver
 
-  constructor({
-    latitudeApiKey,
-    gateway = {
-      host: env.GATEWAY_HOSTNAME,
-      port: env.GATEWAY_PORT,
-      ssl: env.GATEWAY_SSL,
+  constructor(
+    apiKey: string,
+    {
+      projectId,
+      gateway = {
+        host: env.GATEWAY_HOSTNAME,
+        port: env.GATEWAY_PORT,
+        ssl: env.GATEWAY_SSL,
+      },
+    }: {
+      projectId?: number
+      gateway?: GatewayApiConfig
+    } = {
+      gateway: {
+        host: env.GATEWAY_HOSTNAME,
+        port: env.GATEWAY_PORT,
+        ssl: env.GATEWAY_SSL,
+      },
     },
-  }: {
-    latitudeApiKey: string
-    gateway?: GatewayApiConfig
-  }) {
+  ) {
     this.routeResolver = new RouteResolver({
       gateway,
       apiVersion: 'v1',
     })
-    this.latitudeApiKey = latitudeApiKey
+    this.projectId = projectId
+    this.apiKey = apiKey
   }
 
-  async runDocument({
-    params: { projectId, documentPath, commitUuid, parameters, source },
-    onMessage,
-    onFinished,
-    onError,
-  }: {
-    params: {
-      projectId: number
-      documentPath: string
+  async run(
+    path: string,
+    {
+      projectId,
+      commitUuid,
+      parameters,
+      onEvent,
+      onFinished,
+      onError,
+    }: {
+      projectId?: number
       commitUuid?: string
       parameters?: Record<string, unknown>
-      source?: LogSources
+    } & StreamResponseCallbacks,
+  ) {
+    projectId = projectId ?? this.projectId
+    if (!projectId) {
+      onError?.(new Error('Project ID is required'))
+      return
     }
-  } & StreamResponseCallbacks) {
-    const response = await this.makeRequest({
-      method: 'POST',
-      handler: HandlerType.RunDocument,
-      params: { projectId, commitUuid },
-      body: { documentPath, parameters, source },
-    })
 
-    return this.handleStreamChainResponse({
-      response,
-      onMessage,
-      onFinished,
-      onError,
-    })
-  }
+    try {
+      const response = await this.request({
+        method: 'POST',
+        handler: HandlerType.RunDocument,
+        params: { projectId, commitUuid },
+        body: { path, parameters },
+      })
 
-  async addMessages({
-    params: { documentLogUuid, messages, source },
-    onMessage,
-    onFinished,
-    onError,
-  }: {
-    params: {
-      documentLogUuid: string
-      messages: Message[]
-      source?: LogSources
+      return this.handleStream({
+        stream: response.body!,
+        onEvent,
+        onFinished,
+        onError,
+      })
+    } catch (err) {
```

Comment:
run and chat methods should not return a promise that can fail as otherwise it makes the dev experience kind of shit. In theory we should tell devs the way to handle any error from the sdk is through the onError callback.

Later on, we will introduce a no stream option that will return a promise and those methods can actually return _failable_ promises